### ISSUE TITLE
Limit of videos (sockets) per media session

### ISSUE THREAD

**Issue by pagella42** [microsoftgraph/microsoft-graph-comms-samples#229]
Hey, i'm trying to build a bot that can join a Live event, and capture its video-audio stream.

The problem is that so far, we were only able to get as much as 4 videos (4 sockets available).
Is that an actual limit? Is there a socket limit per media session? If so, is it actually 4?

**Comment by ssulzer** [comment]
No, a single bot call may have up to 10 VideoSockets for "main" video + 1 VideoSocket for "vbss" screen-sharing. But there may be fewer than 10 available video sources within the meeting. How many participants were in the Live Event "inner meeting"?

**Comment by pagella42** [comment]
@ssulzer Hey thanks for your fast response!

Could you please help me with this other concern?
I'm trying to have a bot join a live event.

The problem is the bot will always join as a presenter, even if it joins after the event started.
The thing is, i am trying to capture the media from the live event, and as a presenter, i only have access to the internal stream, being the presenters video. But what i'm trying to capture, is the feed that attendees get, the one that the producer chooses.

Is there actually a way to do that? If so, How?
If not, is it a feature on the road map? Is there an ETA?

**Comment by ssulzer** [comment]
@pagella42 
It may be possible to do this. Within the "inner meeting" of the live event (where the producer and presenters and are located, and where your bot also joins), there is another bot that is part of the live event broadcast system, which provides a preview of the video feed that is sent to attendees. Typically the producer's client subscribes to this preview video source, but your bot should be able to obtain it too.
Your bot should be able to find this "composer" bot from the roster - you may actually find two instances of it (one is a backup). It will have a send/receive main video stream (along with receive-only audio and applicationsharing-video streams) - subscribe to that bot's video source and see if it provides what you are looking for.
If you need some help, please provide a callid or scenario-id of a test broadcast meeting that your bot joined. Thanks.

**Comment by stefan2410** [comment]
@ssulzer if we want to limit the recording only in audio, is there a way to not create at all the video/vbss sockets?   I know about the subscribe/unsubscribe to video.

**Comment by ssulzer** [comment]
@stefan2410 
If your bot wants to support only the audio/voice modality, then it can create only an AudioSocket and not any VideoSocket objects. In the sample bots under V1.0Samples/LocalMediaSamples, in the Bot.cs module there is a Bot.CreateLocalMediaSession method which configures whether the bot supports audio or audio & video. If the bot wants to be audio-only, this CreateLocalMediaSession method should not create any VideoSocketSetting objects.

**Comment by stefan2410** [comment]
Thank you very much @ssulzer 
Another problem is that I don't know (don't understand maybe), how we can communicate with the Cloud Services instances. To get from every one the logs or other API calls.  We have some problems with the PIP (or to understand how it works). It seems to me that it is only for the media platform?

**Comment by ssulzer** [comment]
Hi @stefan2410 
Can you clarify your questions please?
What problem do you have with the ILPIPs?
Thanks.

**Comment by stefan2410** [comment]
Hi @ssulzer  we solved the ILPIP's problem. it was a misunderstanding between devs and admins.
Thanks again.
